numpy/core/src/umath/ufunc_object.c

```diff
@@ -4169,15 +4172,23 @@ PyUFunc_GenericReduction(PyUFuncObject *ufunc, PyObject *args,
         return (PyObject *)ret;
     }
 
-    if (Py_TYPE(op) != Py_TYPE(ret)) {
-        res = PyObject_CallMethod(op, "__array_wrap__", "O", ret);
-        if (res == NULL) {
-            PyErr_Clear();
-        }
-        else if (res == Py_None) {
-            Py_DECREF(res);
+    else {
+        PyObject *wrap;
+        _find_array_wrap(args, kwds, &wrap, 1, 1);
+
```

Comment:
Extra blank line above `else`. I'm curious if the `wrap` here matters, the return will already have happened if `out`  is not NULL.